quill-cassandra/src/main/scala/io/getquill/context/cassandra/CassandraSessionContext.scala

```diff
@@ -34,6 +35,20 @@ abstract class CassandraSessionContext[N <: NamingStrategy](
 
   protected val session = cluster.connect(keyspace)
 
+  protected val udtMetadata: Map[String, List[UserType]] = cluster.getMetadata.getKeyspaces.asScala.toList
+    .flatMap(_.getUserTypes.asScala)
+    .groupBy(_.getTypeName)
+
+  protected def udtValueOf(udtName: String): UDTValue = udtMetadata.getOrElse(udtName, Nil) match {
+    case udt :: Nil => udt.newValue()
+    case Nil =>
+      fail(s"Could not find UDT `$udtName` in any keyspace")
+    case udts => udts
+      .find(_.getKeyspace == session.getLoggedKeyspace)
```

Comment:
Behavior is undetermined when several keyspaces contain UDTs with the same name. `find` returns only the first occurence.

Reply:
// Had been commented before I saw response for other note.

Reply:
in this case, udts are grouped by keyspaces, see`udtMetadata`, hence it's impossible to return more than one entries by `_.getKeyspace == session.getLoggedKeyspace` predicate

Reply:
one note, `udtMetadata` is grouped by keyspaces, however cassandra won't let you have multiple udts with the same name in single keyspace